### ISSUE TITLE
Get all children that are Groups (not Arrays)

### ISSUE THREAD

**Issue by niklasmueboe** [LDeakin/zarrs#103]
Hi, 
I was wondering what the best way is to get all children of a Group that are also Groups. Specifically I am only interested in the names of the children.

Currently I am trying this

```
Node::open(&store, "/")?
        .children()
        .into_iter()
        .filter_map(|node| {
            if matches!(node.metadata(), NodeMetadata::Group(_)) {
                Some(node.name().as_str().to_owned())
            } else {
                None
            }
        })
        .collect();
```

but was wondering whether there is a simpler way instead of matching on the metadata type. Also prefereably I would start with a Group already instead of creating a new node.

**Comment by LDeakin** [comment]
That looks to be the way right now.

> Also prefereably I would start with a Group already instead of creating a new node

+1. `Group` would greatly benefit from having methods to get child `Group`s / `Array`s. Would you be interested in contributing such a feature?

**Comment by niklasmueboe** [comment]
I started a draft #104 

But some discussion on what the API should look like might be useful

**Comment by LDeakin** [comment]
Closed by #104. Thanks @niklasmueboe
